// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;

package parameters;

service SegmentRouting {

  rpc AddSrRoute (AddSrRouteRequest) returns (AddSrRouteReply) {}
  rpc RemoveRoute (RmRouteRequest) returns (RmRouteReply) {}
  rpc ListRoute (ListRouteRequest) returns (ListRouteReply) {}
  rpc ChangeRoute (ChangeRouteRequest) returns (ChangeRouteReply) {}
  rpc ChangeSrRoute (ChangeSrRouteRequest) returns (ChangeSrRouteReply) {}
  rpc BunchAddSrRoute (AddSrRouteRequest) returns (AddSrRouteReply) {}
  rpc BunchRemoveRoute (RmRouteRequest) returns (RmRouteReply) {}
  rpc BunchAddSrRouteNew (AddBatchRequest) returns (AddSrRouteReply) {}

}

message AddSrRouteRequest {
  string prefix = 1;
  string encapmode = 2;
  string segments = 3;
  string device = 4;
}

message AddBatchRequest{
  int numberOfEntries = 1;
  string[] encapmode = 2;
  string[] segments = 3;
  string[] device = 4;
}

message AddSrRouteReply {
  string message = 1;
}

message RmRouteRequest {
	string prefix = 1;
	string device = 2;
}

message RmRouteReply {
	string message = 1; 
}

message ListRouteRequest {}

message ListRouteReply {
	string message = 1;
}

message ChangeRouteRequest {
	string prefix = 1;
	string via = 2;
}

message ChangeRouteReply {
	string message = 1;
}

message ChangeSrRouteRequest {
	string prefix = 1;
  	string segments = 2;
  	string device = 3;
}

message ChangeSrRouteReply {
	string message = 1;
}


